horovod/common/common.h

```diff
@@ -153,7 +154,7 @@ namespace common {
 #define JOIN_TENSOR_NAME "join.noname"
 
 // List of supported frameworks.
-enum Framework { TENSORFLOW, PYTORCH, MXNET };
+enum Framework { TENSORFLOW, PYTORCH, MXNET, XLA };
```

Comment:
It seems that XLA is implemented only for tensorflow, why XLA declare here as a framework ? So does the design of cpp namespace level. Thx for your explain in advance.

Reply:
Thanks for the question. My rationale is explained as below:
1. XLA is independent of Tensorflow, although they are in the same repo now. There are talks about separating XLA from Tensorflow although I don't know when that will happen. In addition, XLA (as a DL compiler) is used by frontends other than Tensorflow, such as JAX and Pytorch (not in its main repo, I guess), etc.
2. I certainly can understand that it is inaccurate to say XLA is a framework. However, as XLA is independent of Tensorflow, it does not use Tensorflow constructs and _cannot_ use [TFOpContext](https://github.com/horovod/horovod/blob/0a421945667ab0d914db588bcd48134f48dd0f05/horovod/tensorflow/mpi_ops.cc#L171). It then needs to create a new [XLAOpContext ](https://github.com/horovod/horovod/blob/2337d8180a0c52076ac4224c7a35031fa2af726e/horovod/tensorflow/xla_mpi_ops.cc#L441) with a new "framework" name. I don't see alternatives.
3. If the name is really confusing to people, I may suggest rename "framework" to "backend".

Reply:
Thanks for your reply, you are right that XLA is a layer independent of tf or even torch. XLA is a layer tf/torch can leverage on in my opinion, that's why I propose the question. I'm not sure am I right with following consideration,
1. XLA can not be used/run without a framework(tf/torch/mxnet)
2. XLA will be run in the way tf+XLA (the version you are working on), or torch+XLA, or more.

If so, may be we should keep framework declaration and consider to implement XLA without changing it even by add *backend* ?

Overall, your work is wonderful, it's ok to keep this version if we cannot figure out a better one.

Reply:
You are right that XLA itself is not complete as a framework.

How about let's rename `XLA` to `TF_XLA`?
That is, `enum Framework { TENSORFLOW, PYTORCH, MXNET, TF_XLA };`

This is conceptually clean and simple. The only theoretical drawback I can think of for this approach is that you might later have PYTORCH_XLA, etc. due to combination (of framework and XLA) but I doubt that if this will really become a problem in practice (mainly due to few combinations that really exist in the world). Even if this becomes a problem in practice in the future, we will know better how to deal with it at that time given more data points.

I will make the change accordingly if this makes sense to you. Please let me know.

Reply:
You are right that XLA itself is not complete as a framework.

How about let's rename `XLA` to `TF_XLA`?
That is, `enum Framework { TENSORFLOW, PYTORCH, MXNET, TF_XLA };`

This is conceptually clean and simple. The only theoretical drawback I can think of for this approach is that you might later have PYTORCH_XLA, etc. due to combination (of framework and XLA) but I doubt that if this will really become a problem in practice (mainly due to few combinations that really exist in the world). Even if this becomes a problem in practice in the future, we will know better how to deal with it at that time given more data points.

I will make the change accordingly if this makes sense to you. Please let me know.

Reply:
Great, we have clarity on the issues but not achieve a solution yet, maybe we should discuss it with the community.

/cc @maxhgerlach  @EnricoMi